.github/workflows/check-commits.yml

```diff
@@ -61,3 +62,26 @@ jobs:
               repo: context.repo.repo,
               body: output
             })
+  # check that only released Gradle versions are used in the wrapper
+  check_wrapper:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check used wrapper
+        run: |
+          set -eu
+          for commit in $(git rev-list ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}); do
+              git checkout $commit
+              WRAPPER_GRADLE_VERSION=$(grep "distributionUrl" gradle/wrapper/gradle-wrapper.properties | sed 's/.*gradle-\(.*\)-[a-z]*\.[a-z]*/\1/')
+              echo "Commit $commit wrapper: $WRAPPER_GRADLE_VERSION"
+              if ! [[ $WRAPPER_GRADLE_VERSION =~ ^[0-9.]+(-(rc|milestone|m)-[0-9]+)?$ ]]; then
+                  echo "Bad wrapper version $WRAPPER_GRADLE_VERSION used in commit $commit. Please rebase your branch to ensure that each commit uses only released Gradle versions in wrapper (GA, RC or milestone)."
```

Comment:
💅 This output should be sent to the error output (I assume GH will then print the log line in red making the error message more visible)

Reply:
Unfortunately, stderr is not colored in red. But I've added some color
https://github.com/gradle/gradle/actions/runs/12669360455/job/35306716619